### ISSUE TITLE
Combine into a single definition file?

### ISSUE THREAD

**Issue by vincerubinetti** [jhwz/vite-plugin-typed-css-modules#3]
I like the idea of this, but it ends up vastly increasing the amount of files in my repo. Now for every component, I have not just one `.tsx` file, but the `.module.css` and now the `module.css.d.ts` file. Would it somehow be possible to combine this into a single definitions file and scope each definition to a particular import.

Also a side note, perhaps this should generate the definition files on first start too, instead of just on file save, for people who are running it for the first time.

Also in the read, it should probably be `--dev` in the `yarn add` command.

**Comment by jhwz** [comment]
Not easily, this repository is just a simple wrapper around https://github.com/Quramy/typed-css-modules.

If you're worried about clutter and use VSCode, I would recommend looking at the `explorer.fileNesting` settings. You can easily nest those files under the `*.module.css` files.

> Also a side note, perhaps this should generate the definition files on first start too, instead of just on file save, for people who are running it for the first time.
>  Also in the readme, it should probably be --dev in the yarn add command.

I agree, good points. I think I also need to handle the case where the `.css.module` file is deleted.

**Comment by vincerubinetti** [comment]
Yeah I didn't realize this was based on that, I stumbled upon that one afterward. I think maybe this issue is requesting what I'm requesting: https://github.com/Quramy/typed-css-modules/issues/27

Thanks for the tip about `fileNesting`, I'll consider it.

**Comment by jhwz** [comment]
> I like the idea of this, but it ends up vastly increasing the amount of files in my repo. Now for every component, I have not just one .tsx file, but the .module.css and now the module.css.d.ts file. Would it somehow be possible to combine this into a single definitions file and scope each definition to a particular import.

As now mentioned in [`Usage`](https://github.com/jhwz/vite-plugin-typed-css-modules#usage) section of the README we recommend adding the `*. module.css.dts` files to your `.gitignore` file. We will generate them when you either start the dev server or build your app.

> Also a side note, perhaps this should generate the definition files on first start too, instead of just on file save, for people who are running it for the first time.

This is done thanks to @KATT in https://github.com/jhwz/vite-plugin-typed-css-modules/pull/5
